File: pisio-frontend/main/src/app/models/incident.ts
import { IncidentSubtype } from "./incident_subtype";
import { IncidentType } from "./incident_type";
import {Location} from "./location";


export interface Incident{

    id: number | null; 
    description : string; 
    incidentType: IncidentType; 
    incidentSubtype : IncidentSubtype; 
    photoLink: string | null; 
    longitude : number; 
    latitude: number;
    approved: number;
    dateOfReport: string;

}

File: pisio-backend/incident_report/src/main/java/com/example/pisio/incident_report/models/Incident.java
package com.example.pisio.incident_report.models;

import com.example.pisio.incident_report.models.entities.IncidentSubtypeEntity;
import com.example.pisio.incident_report.models.entities.LocationEntity;
import com.example.pisio.incident_report.models.entities.ModeratorEntity;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
@Data
public class Incident {


    private Integer id;
    private String description;
    private String photoLink;
    private String dateOfReport;
    //private LocalDate dateOfApproval;
    private Boolean approved;
    private Location location;
    //private ModeratorEntity moderator;
    private IncidentSubtype incidentSubtype;
    //private IncidentSubtypeEntity incidentSubtype;
}
